### PR TITLE
windows development: info about module_utils and requires

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Adds info about `#Requires` and `#AnsibleRequires` as well as how to use module_utils.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
windows development

##### ANSIBLE VERSION
```
2.5
```

### CODE REVIEW

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Edited. Docs are good, but would really be better if they were fleshed out with specific command line and code examples rather than just high-level bullet points. Thanks @jborean93 - nice work!

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst [hunk 1]
```diff
@@ -132,42 +200,41 @@ the most popular are
...
+- Navigate to ``%TEMP%\..``. It should contain a folder starting with ``ansible-tmp-``.
+- Inside this folder, open the PowerShell script for the module.
+- In this script is a raw JSON script under ``$json_raw`` which contains the module arguments under ``module_args``. These args can be assigned manually to the ``$complex_args`` variable that is defined on your debug script.
 
 
 Windows unit testing
 ====================
 
 Currently there is no mechanism to run unit tests for Powershell modules under Ansible CI.
-There is work in the pipeline to introduce this in the future, stay tuned.
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/developing_modules_general_windows.rst:147]: Removed. Please do not discuss possible upcoming features in the documentation. Feel free to ping me for further explanation.

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst [hunk 2]
```diff
@@ -60,14 +60,82 @@ is the order to favour when writing modules:
...
+The following is a list of module_utils that are packaged with Ansible and a general description of what
+they do:
+
+- ArgvParser: Utiliy used to convert a list of arguments to an escaped string compliant with the Windows argument parsing rules.
+- CamelConversion: Utility used to convert camelCase strings/lists/dicts to snake_case.
+- CommandUtil: Utility used to execute a Windows process and return the stdout/stderr and rc as separate objects.
+- Legacy: General definitions and helper utilities for Ansible module.
+- SID: Utilities used to convert a user or group to a Windows SID and vice versa.
+
+For more details on any specific module utility and their requirements, please see the `Appendix: Ansible Module Utilities <http://docs.ansible.com/ansible/latest/dev_guide/developing_module_utilities.html>_` and the `Ansible
```
Comment (jborean93) [inline docs/docsite/rst/dev_guide/developing_modules_general_windows.rst]: The link to the `developing_module_utilities` isn't really relevant to the PowerShell module_utils and probably will lead to confusing, do you mind if I remove it @dharmabumstead

**Review by jborean93 — COMMENTED** [review]
@dharmabumstead just one question regarding your changes. As for fleshing out the content we can always do that in time but my aim is to just get at least something out there for people to get started on.

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst (hunk 2 above)
Comment (dharmabumstead) [reply docs/docsite/rst/dev_guide/developing_modules_general_windows.rst]: Sure.

**Pull request merged**